### DISCUSSION TITLE
Signing the builds when building locally

### DISCUSSION THREAD

**Discussion by wir3z** [Snaggly/KSW-ToolKit#21] [General]
Great app BTW! I was playing with building it locally, and there is this note you posted "When compiling, you will need to sign the APK with the same KeyStore as the Android OS was signed with."

Can you explain what exactly that means?  Do I just compile with the same SDK version as the host it will run on?  Or do I need to get the key store off the device I wish to deploy on?  Thanks!

**Comment by Snaggly** [comment]
Hello,
the Apk just needs to be signed with the same key as the Apks found under /system/app or /system/priv-app. The reason for that is that it requires a permission that is only granted for apks with the same Signature as the system apps.

**Reply by wir3z** [reply]
That makes sense.  Where do I find that key?

**Reply by Snaggly** [reply]
The Firmware builders would have them. Unfortunately it's not possible to reverse the private/public keyset from just the signature. I cannot help you more in this I'm afraid.

**Reply by wir3z** [reply]
Understood!  Thanks!
